cli/base.ts

```diff
@@ -25,4 +25,17 @@ export abstract class BaseCommand extends Command {
     })
     this.log(outputDetails, ' output to file: ', filename)
   }
+  
+  // Determine block number from user parameters. If a date was submitted, use date to calculate 
+  // corresponding block number. If no date, use the block number submitted.
+  async determineBlockNumber(block: number | undefined, date: string | undefined, web3: Web3): Promise<number | undefined> {
```

Comment:
Are you sure that this approximation is good enough? While Celo's blocktime has been pretty darn good, I'm not sure it actually has been on average exactly 5s

Reply:
I initially rejected this idea. However @asaj said we don't need to be extremely exact, and prestwich said it's a pretty good estimate. I've tested out a day in each month, and I always get a block number from that particular day (within 30 min of each other, from April to Jan the time it lands on has slowly increased by 30 min). The biggest downside is that if the average ever changes this needs to be changed. Also the time lands pretty much at the start of the day around 1am EST. So the upper bound is exclusive here. I can either document that or add a day to the given date so that it ends around midnight of the following "day".

Maybe long term, we can do something like request blocks and do a binary search until we find our desired timestamp. Given the tight deadline I thought this worked just fine for now. Since the product team is interested in using this tool do get balances for certain weeks, I thought this parameter would just make things a little more convenient for now. (But it would be great if metabase could fulfill this need for them somehow 0_o) lol.

Reply:
Sounds great! Yeah would be nice to have an indication that this is an approximation, but totally reasonable to make that trade-off

Reply:
I'll file some issues with this repo, for anything like this that comes up